SearchManager.h

```diff
@@ -0,0 +1,37 @@
+#ifndef TICTACTOE_SERVER_SEARCHMANAGER_H
+#define TICTACTOE_SERVER_SEARCHMANAGER_H
+
+#include <unordered_map>
+#include <random>
+
+class SearchManager {
+public:
+	struct PlayerInfo {
+		std::string id;
+		std::string adress;
+	};
+
+private:
+	unordered_map<unsigned, PlayerInfo> active_searches;
+
+public:
+	
+	bool check_for_active_search( PlayerInfo player) {
+		if (active_searches.empty()) {
+			active_searches[std::mt19937(std::random_device()())()] = player ;
+			return false;
+		} else {
+			return true;
+		}
+	}
+
+	PlayerInfo get_player_info() {
```

Comment:
- This is a compile-time error. A map's iterator's value is a pair (key + value), not just the value
- Why do you only need the player info without the search id?

Reply:
This isn't resolved.